### PR TITLE
[Merged by Bors] - Add a random access get_component benchmark

### PR DESCRIPTION
**Pull request by TheRawMeatball**
# Objective

Add a benchmark to measure the performance of get_component, particularly for cases involving random access.

Enables #2965

### CODE REVIEW

**Review by james7132 — APPROVED** [review]

File: benches/benches/bevy_ecs/world_get.rs [hunk 1]
```diff
@@ -20,6 +28,10 @@ struct Sparse(f32);
 
 const RANGE: std::ops::Range<u32> = 5..6;
```
Comment (alice-i-cecile) [inline benches/benches/bevy_ecs/world_get.rs:29]: This being set to 5..6 is a little weird. Not introduced here, but it really feels like this should be something more like 1..10.

I'd like to get a larger curve, and test at smaller entity counts too.

**Review by alice-i-cecile — APPROVED** [review]
I have a quibble about the range of values that these benches are running over, but the bench itself LGTM and is very important.

File: benches/benches/bevy_ecs/world_get.rs (hunk 1 above)
Comment (TheRawMeatball) [reply benches/benches/bevy_ecs/world_get.rs:29]: I'd prefer leaving these sorts of changes to a separate PR.

Comment (mockersf) [reply benches/benches/bevy_ecs/world_get.rs:29]: 1..10 would make the benches a lot longer to run

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
Actually, one blocking request. Can you also bench `Query::get` fetching both `Sparse` and `Table`? IIRC Cart told me ages ago that this had different performance characteristics from get_component, and I'd like to make sure we avoid major perf regressions to compound gets.

**Review by mockersf — APPROVED** [review]

**Pull request closed without merging**